hpx/parallel/executors/distribution_policy_executor.hpp

```diff
@@ -60,53 +72,54 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v3)
                 "distribution policy type");
 
         // apply_execute implementations
-        template <typename F>
+        template <typename F, typename ... Ts>
         typename std::enable_if<
             !hpx::traits::is_action<F>::value
         >::type
-        apply_execute(F && f) const
+        apply_execute_impl(F && f, Ts && ... ts) const
         {
             typedef components::server::invoke_function_action<F> action_type;
-            policy_.template apply<action_type>(
-                threads::thread_priority_default, std::forward<F>(f));
+            policy_.template apply<action_type>(threads::thread_priority_default,
+                std::forward<F>(f), std::forward<Ts>(ts)...);
         }
 
 #if !defined(HPX_GCC_VERSION) || HPX_GCC_VERSION >= 40700
-        template <typename Action>
+        template <typename Action, typename ... Ts>
         typename std::enable_if<
             hpx::traits::is_action<Action>::value
         >::type
-        apply_execute(Action && act) const
+        apply_execute_impl(Action && act, Ts && ... ts) const
         {
-            policy_.template apply<Action>(
-                threads::thread_priority_default);
+            policy_.template apply<Action>(threads::thread_priority_default,
+                std::forward<Ts>(ts)...);
         }
 #endif
 
         // async_execute implementations
-        template <typename F>
+        template <typename F, typename ... Ts>
         typename std::enable_if<
-            !hpx::traits::is_action<F>::value,
+           !hpx::traits::is_action<F>::value,
             hpx::future<
                 typename detail::distribution_policy_execute_result<F>::type
             >
         >::type
```

Comment:
That's what I'm questioning here (though unrelated to that PR): does it make sense to call non-actions with distribution policies?